pkg/source/resource_test.go

```diff
@@ -0,0 +1,131 @@
+package source_test
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"testing"
+
+	diecorev1 "dies.dev/apis/core/v1"
+	diemetav1 "dies.dev/apis/meta/v1"
+	"github.com/spf13/cobra"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/cli-runtime/pkg/resource"
+	"k8s.io/client-go/rest/fake"
+	"k8s.io/client-go/restmapper"
+	k8sscheme "k8s.io/kubectl/pkg/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	cartov1alpha1 "github.com/vmware-tanzu/apps-cli-plugin/pkg/apis/cartographer/v1alpha1"
+	"github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime"
+	clitesting "github.com/vmware-tanzu/apps-cli-plugin/pkg/cli-runtime/testing"
+	"github.com/vmware-tanzu/apps-cli-plugin/pkg/commands"
+	"github.com/vmware-tanzu/apps-cli-plugin/pkg/flags"
+	"github.com/vmware-tanzu/apps-cli-plugin/pkg/source"
+)
+
+func TestWorkloadOptionsFetchResourceObject(t *testing.T) {
```

Comment:
func `FetchResourceObject`  has 3 output scenario
- when fetch fails
- when fetch succeeds, no pods
- when fetch succeeds and some pods

All 3 cases should be covered

Reply:
Updated the code to address these